pkg/clientcfg/clientcfghttp/handler.go

```diff
@@ -112,7 +112,7 @@ func (h *HTTPHandler) RegisterRoutes(router *mux.Router) {
 func (h *HTTPHandler) serviceFromRequest(w http.ResponseWriter, r *http.Request) (string, error) {
 	services := r.URL.Query()["service"]
 	if len(services) != 1 {
-		h.metrics.BadRequest.Inc(1)
+		// h.metrics.BadRequest.Inc(1)
```

Comment:
TODO: throwing error, because I am using null metric factory.

Reply:
are you passing `nil` or `metrics.NullFactory`? The latter does not throw (it's just a no-op implementation).